config/puma.rb

```diff
@@ -0,0 +1,36 @@
+# Adjust setting below to your local environment and than start puma like this:
+#	puma -C config/puma.rb --control=unix:///tmp/puma.ctrl --control-token=
+# and manage it like this:
+#	pumactl --control-url=unix:///tmp/puma.ctrl {restart|stop|status|stats|halt}
+# For more information see http://www.rubydoc.info/gems/puma/
+
+environment 'production'
+# For non-thread-capable ruby impl. like MRI modifying this is usually just
+# a waste of resources. However, for Rubinius or JRuby it probably makes a
+# difference - in this case don't forget to adjust the workers accordingly.
+threads 1,1
+# Perhaps starting with 75% of available strands is ok - monitor and adjust.
+workers 24
+# For convinience make it the same as for 'rails server', i.e. listen on all
+# available interfaces on port 3000
+bind 'tcp://0.0.0.0:3000'
+# We don't like noise
+quiet
+# IMPORTANT wrt. MRI ruby - saves rsources!
+preload_app!
+
+# let's make sure, that ppl with the same GID as the running puma process
+# are really able to control puma (the group needs write access). See option
+# --control and --control-url above.
+on_worker_boot do
+	str = @options[:control_url]
```

Comment:
The given URL is obviously unrelated and doesn't proof, that it contains "more" than just a string. Anyway, it is a riddle to me, why we should waste our valuable time with such "out-of-the-scope" stuff: If there is no API, one can't rely on anything unless the complete application got analyzed and one can be 100% certain, that its assumption is right - sorry, but I've no time for such nonsense!  When a new version comes in, one has to do it again - sorry again, I've no time for such nonsense.  And last but not least and already said, the web application is using rails and thus this is the only concern one has to deal with. It is beyond the scope of an web app using a certain framework to dictate, which exact web server should be used, and AFAICS for passenger I can't see any similar constraints. And yes, strictly taken, this doesn't belong into the web app, it just provides out-of-the-box support (I didn't make the rules, how ruby apps get assembled/packaged ...). So sink or swim, take or discard - these are my last words on this - already wasted too much time on this.